### PR TITLE
Base Entities 

### PR DESCRIPTION
**Pull request by Amonov996**
Add Base Entities,
Add NotificationDbContext
Add EntitiyRepositoryBase
Install NuGet EFCore on Persistence

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+using Microsoft.EntityFrameworkCore;
+
+namespace AirBnB.Persistence.Repositories;
+
+public abstract class EntityRepositoryBase<TEntity, TContext> where TEntity : class, IEntity where TContext : DbContext
+{
+    protected TContext DbContext => (TContext) _dbContext;
+    private readonly DbContext _dbContext;
+
+    protected EntityRepositoryBase(DbContext dbContext)
```
Comment (boburjon1504) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs:12]: Shu yerida construktorni o'zidan TContextni bersez ortiqcha castga hojat qolmendi

Comment (Amonov996) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs:12]: Qanday qilib ? Misol keltira olasizmi ?

**Review by Ilxom-22 — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/CreationAuditableEntity.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public class CreationAuditableEntity : ICreationAuditableEntity
+{
+    
+}
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/CreationAuditableEntity.cs:6]: qo'shimcha 3 ta ochilgan iterfacelarga class bo'lishi shart emas shunchaki interfaceni o'zi bo'lsa yetarli yani CreationAuditableEntity shart emas interfaceni o'zi yetarli

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ICreationAuditableEntity.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface ICreationAuditableEntity
+{
+    
+}
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ICreationAuditableEntity.cs:6]: buni bitta 
Guid CreatedBy property si bo'lishi kerak

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IDeletionAuditableEntity.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface IDeletionAuditableEntity : IAuditableEntity
+{
+    
+}
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IDeletionAuditableEntity.cs:6]: Bu AuditableEntitydan olishi shart emas
va bunda ham bittta DeletedBy propertysi bo'lishi kerak Guid tipida

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IModificationAuditableEntity.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface IModificationAuditableEntity
+{
+    
+}
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IModificationAuditableEntity.cs:6]: Bunda ham bitta ModifiedBy property bo'ladi Guid tipida

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ISoftDeletedEntity.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+namespace AirBnB.Domain.Common;
+
+public interface ISoftDeletedEntity : IEntity 
+{
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ISoftDeletedEntity.cs:4]: ISoftDeleted faqatgina IAuditabledan olsa yeta

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/SoftDeletedEntity.cs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+namespace AirBnB.Domain.Common;
+
+public class SoftDeletedEntity : ISoftDeletedEntity
+{
+    public Guid Id { get; set; }
```
Comment (Abdurahmon0412) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/SoftDeletedEntity.cs:5]: SoftDeletedEntity : AuditableEntity, ISoftDeletedEntity
softdeletedni o'zi mana shunay oladi

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs (hunk 1 above)
Comment (Abdurahmon0412) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs:12]: pretocted constructorda DbContext dbContext deb emas balki TContext dbContext deb qabul qilasiz va keyin tepadagi strelkadan keyingi qavsni o'chirib tashlasayiz ham bo'ladi

**Review by Abdurahmon0412 — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/Entity.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public class Entity : IEntity
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/Entity.cs:3]: Common classlar abstract class bo'lishi kerak.

**Review by Ilxom-22 — CHANGES_REQUESTED** [review]

**Review by Abdurahmon0412 — APPROVED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ICreationAuditableEntity.cs [hunk 9]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface ICreationAuditableEntity
+{
+    public Guid CreatedBy { get; set; }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ICreationAuditableEntity.cs:5]: Update to `CreatedByUserId` and add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IDeletionAuditableEntity.cs [hunk 10]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface IDeletionAuditableEntity 
+{
+    public Guid DeletedBy { get; set; }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IDeletionAuditableEntity.cs:5]: Update to `DeletedByUserId` and add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IAuditableEntity.cs [hunk 11]
```diff
@@ -0,0 +1,8 @@
+namespace AirBnB.Domain.Common;
+
+public interface IAuditableEntity : ISoftDeletedEntity
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IAuditableEntity.cs:3]: Add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/Entity.cs [hunk 12]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public abstract class Entity : IEntity
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/Entity.cs:3]: Add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/AuditableEntity.cs [hunk 13]
```diff
@@ -0,0 +1,8 @@
+namespace AirBnB.Domain.Common;
+
+public abstract class AuditableEntity : SoftDeletedEntity, IAuditableEntity
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/AuditableEntity.cs:3]: Add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IEntity.cs [hunk 14]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface IEntity
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IEntity.cs:3]: Add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ISoftDeletedEntity.cs [hunk 15]
```diff
@@ -0,0 +1,8 @@
+namespace AirBnB.Domain.Common;
+
+public interface ISoftDeletedEntity : IEntity
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/ISoftDeletedEntity.cs:3]: Add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IModificationAuditableEntity.cs [hunk 16]
```diff
@@ -0,0 +1,6 @@
+namespace AirBnB.Domain.Common;
+
+public interface IModificationAuditableEntity
+{
+    public Guid ModifiedBy { get; set; }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Common/IModificationAuditableEntity.cs:5]: Update to `ModifiedByUserId` and add docs

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/DataContexts/NotificationDbContext.cs [hunk 17]
```diff
@@ -0,0 +1,16 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace AirBnB.Persistence.DataContexts;
+
+public class NotificationDbContext : DbContext
+{
+    public NotificationDbContext(DbContextOptions<NotificationDbContext> options) : base(options)
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/DataContexts/NotificationDbContext.cs:7]: Use primary constructor, update projects to .net 8

File: src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs [hunk 18]
```diff
@@ -0,0 +1,119 @@
+using System.Linq.Expressions;
+using AirBnB.Domain.Common;
+using Microsoft.EntityFrameworkCore;
+
+namespace AirBnB.Persistence.Repositories;
+
+public abstract class EntityRepositoryBase<TEntity, TContext> where TEntity : class, IEntity where TContext : DbContext
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Persistence/Repositories/EntityRepositoryBase.cs:7]: Add docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]
Good job

**Pull request merged**